### DISCUSSION TITLE
HelloCoroutine on Platformio on Digispark (Attiny85) fails compile

### DISCUSSION THREAD

**Discussion by mkarliner** [bxparks/AceRoutine#58] [General]
I've tried compiling the HelloCoroutine code on Platformio for the Digispark board, which is attiny85 based.
I get the following compile errors

Demo repo is here: git@github.com:mkarliner/AceTest.git

```
In file included from .pio/libdeps/digispark-tiny/AceCommon/src/AceCommon.h:55:0,
                 from .pio/libdeps/digispark-tiny/AceRoutine/src/ace_routine/Coroutine.h:30,
                 from .pio/libdeps/digispark-tiny/AceRoutine/src/AceRoutine.h:49,
                 from src/main.cpp:8:
.pio/libdeps/digispark-tiny/AceCommon/src/print_str/PrintStr.h:130:10: error: 'void ace_common::PrintStrBase::flush()' marked 'override', but does not override
     void flush() override {
          ^~~~~
Compiling .pio/build/digispark-tiny/libfea/AceRoutine/ace_routine/Coroutine.cpp.o
*** [.pio/build/digispark-tiny/src/main.cpp.o] Error 1
In file included from .pio/libdeps/digispark-tiny/AceCommon/src/AceCommon.h:55:0,
                 from .pio/libdeps/digispark-tiny/AceRoutine/src/ace_routine/Coroutine.h:30,
                 from .pio/libdeps/digispark-tiny/AceRoutine/src/ace_routine/Coroutine.cpp:25:
.pio/libdeps/digispark-tiny/AceCommon/src/print_str/PrintStr.h:130:10: error: 'void ace_common::PrintStrBase::flush()' marked 'override', but does not override
     void flush() override {
          ^~~~~
*** [.pio/build/digispark-tiny/libfea/AceRoutine/ace_routine/Coroutine.cpp.o] Error 1

```

**Comment by bxparks** [comment]
I don't support PlatformIO so you have to work a little harder. If you trace the code into AceCommon, you'll see that there are a bunch of `#ifdef` macros for the ATtiny85. Apparently PlatformIO does not define those macros. If you find out what those macros are, perhaps you can submit a PR to the AceCommon project, just like this one which solves a very similar problem: https://github.com/bxparks/AceCommon/pull/28

[Converting this into a Discussion, because it's _not_ a bug with AceRoutine.]

**Reply by mkarliner** [reply]
My apologies Brian.  Your readme says that AceRoutine "should" support PlatformIO but you hadn't tested it, so I thought you'd welcome a PlatformIO related issue. 
I'll have a think about a good way to do this (excuse me if my pull requests might be rubbish, I don't do this often enough).
I wonder if I should raise an issue with TinyCore to make a clearer, cleaner solution. All they have to do is make a TinyCore define, as I understand it.

**Reply by bxparks** [reply]
I guess it could be interpreted that way, but I generally don't support things that I don't test myself. I don't go out of my way to break PlatformIO, but I cannot handle supporting 2X the number of combinations of hardware, libraries, and dev tools. With regards to TinyCore, I don't know how PlatformIO integrates TinyCore, and I don't know why those macros aren't defined in PlatformIO. You'll have to chase after that yourself. In other words, send me solutions, not just problems.
